int16 id
float32 x
float32 y
---
string msg